### DISCUSSION TITLE
Trying a DIGIT Quickstart deploy on AWS

### DISCUSSION THREAD

**Discussion by subhavasan** [egovernments/DIGIT-DevOps#887] [Q&A]
Following this doc for Digit Quickstart deploy on AWS --> https://docs.digit.org/v/v2.6/setup/quickstart/1.-infra-setup

Using the eGov AWS account to test this out. Followed all instructions as per documentation. Could not install terraform 0.14.10 because that specific version is unavailable for Mac M1. Installed latest terraform using home-brew:

`brew install terraform`. 

Version installed was 1.2.

I am getting the following error when I run ```terraform plan```:

```
subha@Subhashinis-MacBook-Air quickstart-aws-ec2 % terraform plan
╷
│ Error: error configuring Terraform AWS Provider: error validating provider credentials: error calling sts:GetCallerIdentity: operation error STS: GetCallerIdentity, https response error StatusCode: 403, RequestID: 24629229-5a39-4580-9953-4ae58761f7d6, api error ExpiredToken: The security token included in the request is expired
│ 
│   with provider["registry.terraform.io/hashicorp/aws"],
│   on main.tf line 1, in provider "aws":
│    1: provider "aws" {
```

**Comment by gajenc** [comment]
AWS access is timebound, meaning your access token will expire in 4 hours. If you have configured the AWS profile with the session token earlier than 4 hours of time, update the new session token which you can get it from the same link.

**Reply by subhavasan** [reply]
For eGov employees: I had to get the administrator access to the AWS eGov env and corresponding tokens to get this to work. The user has to be added to some ACL for admin access.

**Reply by subhavasan** [reply]
@gajendran-egov - I enabled port forwarding on my MacBook Air. Downloaded the postman collection and attempted running the collection. All pods on the cluster are running just fine. Postman error looks like this:

<img width="1280" alt="Screenshot 2022-06-07 at 3 48 38 PM" src="https://user-images.githubusercontent.com/4337305/172356969-06d00ff2-e7f8-4c65-9aff-b8f75fa30724.png">

Getting an error, see details below. For this particular failure case, the hostname is Quickstart.local.digit whereas for the earlier ones, it was pointing to 127.0.0.1:8080. Is that perhaps the issue?

```
POST http://quickstart.local.digit/user/oauth/token
Error: connect ECONNREFUSED 127.0.0.1:80
Request Headers
authority: quickstart.local.digit
sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"
accept: application/json, text/plain, */*
authorization: Basic ZWdvdi11c2VyLWNsaWVudDo=
sec-ch-ua-mobile: ?0
user-agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36
content-type: application/x-www-form-urlencoded
origin: http://quickstart.local.digit
sec-fetch-site: same-origin
sec-fetch-mode: cors
sec-fetch-dest: empty
referer: http://quickstart.local.digit/employee/user/login
accept-language: en-GB,en;q=0.9
Postman-Token: fa7dd6bf-fd78-4748-94fa-197f0709a3c4
Host: quickstart.local.digit
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
Request Body
username: "GRO"
password: "eGov@4321"
grant_type: "password"
scope: "read"
tenantId: "pg.citya"
userType: "EMPLOYEE"
```

**Reply by subhavasan** [reply]
Found the issue. All API requests in the collection need to be 127.0.0.1:8080 and not have quickstart.local.digit. Postman collection needs to be modified. 

Having a collection which deletes all the created users and restores the system back to ground zero will be useful. That way, these tests can be run repeatedly for testing purposes.
